src/worker/tasks/chainIndexer.ts

```diff
@@ -332,7 +351,43 @@ export const createChainIndexerTask = async (args: {
             }),
           ]);
 
-          // update the block number
+          // Map { contractAddress => array of webhooks }
```

Comment:
This is where contract subscriptions webhooks are enqueued.

Pseudocode
```
get logs and receipts that were inserted to db
get a map of (address => webhooks)
for each log
  send to all webhooks for that contract address, if any
for each receipt
  send to all webhooks for that contract address, if any
```

If something throws before all webhooks are enqueued, we fail and not update `currentBlockNumber`.